gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenMetrics.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.home;
+
+import android.graphics.drawable.Drawable;
+import android.view.View;
+import androidx.annotation.Nullable;
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
+import com.google.android.gnd.R;
+import com.google.android.gnd.ui.common.BottomSheetBehavior;
+
+public class HomeScreenMetrics {
+  private final CoordinatorLayout parent;
+  private final View view;
+
+  public HomeScreenMetrics(CoordinatorLayout parent, View view) {
+    this.parent = parent;
+    this.view = view;
+  }
+
+  public static float scale(float value, float before1, float before2, float after1, float after2) {
```

Comment:
Since these are all public methods it might be useful to future readers to add a comment to each one explaining what it does (even if it's obvious from the method name) and the context in which it's likely to be used

Reply:
I've actually rewritten this class completely to use pixel values instead of ratios for anchor points. That version has JavadDoc.